src/components/ProjectSettingsTabs/PluginsTab/getFormSchema.ts

```diff
@@ -0,0 +1,189 @@
+import { Field } from "@rjsf/core";
+import { FormDataProps, SpruceFormProps } from "components/SpruceForm";
+import { CardFieldTemplate } from "components/SpruceForm/FieldTemplates";
+import widgets from "components/SpruceForm/Widgets";
+import { hiddenIf, placeholderIf, radioBoxOptions } from "../utils";
+import { FormState } from "./types";
+
+export const getFormSchema = (
+  useRepoSettings: boolean,
+  repoData?: FormState,
+  formData?: FormDataProps
+): {
+  fields: Record<string, Field>;
+  schema: SpruceFormProps["schema"];
+  uiSchema: SpruceFormProps["uiSchema"];
+} => ({
+  fields: {},
+  schema: {
+    type: "object" as "object",
+    properties: {
+      performanceSettings: {
+        type: "object" as "object",
+        title: "Performance Plugins",
+        properties: {
+          perfEnabled: {
+            type: ["boolean", "null"],
+            title: "",
+            oneOf: radioBoxOptions(
+              ["Enabled", "Disabled"],
+              repoData?.performanceSettings?.perfEnabled
+            ),
+          },
+        },
+      },
+      buildBaronSettings: {
+        type: "object" as "object",
+        title: "Build Baron and Task Annotations",
+        properties: {
+          taskAnnotationSettings: {
+            title: "",
+            type: "object" as "object",
+            properties: {
+              jiraCustomFields: {
+                type: "array" as "array",
+                title: "Custom Jira Fields",
+                items: {
+                  type: "object" as "object",
+                  properties: {
+                    field: {
+                      type: "string" as "string",
+                      title: "Field",
+                    },
+                    displayText: {
+                      type: "string" as "string",
+                      title: "Display Text",
```

Comment:
Should display text be required if Field is populated?

Reply:
I think it would be more of a nice to have. I don't think it needs to be required. Especially since it's very possible that some fields will have human-readable names (like "epic" instead of custom_1234).